appinventor/blocklyeditor/src/field_lexical_variable.js

```diff
@@ -100,19 +102,25 @@ Blockly.FieldLexicalVariable.prototype.setValue = function(text) {
  * Update the eventparam mutation associated with the field's source block.
  */
 Blockly.FieldLexicalVariable.prototype.updateMutation = function() {
+  // TODO: I think I need someone to explain this function to me.
+
   var text = this.getText();
+  // TODO: Change to if return;
   if (this.sourceBlock_ && this.sourceBlock_.getParent()) {
     this.sourceBlock_.eventparam = undefined;
+    // TODO: Is this necessary? I don't think global vars can have parents.
```

Comment:
In context, the this,sourceBlock_ is either a lexical_variable_set or lexical_variable_get block, which are not necessarily top level blocks.

Reply:
Ok that makes way more sense than what I was thinking lol